doc/ipython-notebooks/multiclass/KNN.ipynb

```diff
@@ -260,14 +260,21 @@
     "knn.set_k(3)\n",
     "knn.set_knn_solver_type(KNN_COVER_TREE)\n",
     "pred = knn.apply_multiclass(feats_test)\n",
-    "print \"Covertree KNN took %2.1fs\" % (Time.get_curtime() - start)\n"
+    "print \"Covertree KNN took %2.1fs\" % (Time.get_curtime() - start)\n",
+    "\n",
+    "\n",
+    "start = Time.get_curtime()\n",
+    "knn.set_k(3)\n",
+    "knn.set_knn_solver_type(KNN_KDTREE)\n",
+    "pred = knn.apply_multiclass(feats_test)\n",
+    "print \"KDTree KNN took %2.1fs\" % (Time.get_curtime() - start)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "So we can significantly speed it up. Let's do a more systematic comparison. For that a helper function is defined to run the evaluation for KNN:"
+    "KNN predictions are very costly: predictiong a new point's class requires computing distances to all training samples. While Shogun exploits multi-core infrastructure, KNN quickly becomes infeasible for large datasets. We can see it been significantly speed up by Cover Trees solver, but the result of KD-Trees solver seems like not so ideally. We now do a more systematic comparison of Shogun's KNN implementations. We begin by defining a helper function:"
```

Comment:
You just need to say what you mean in a clean and neat way, and also avoid grammar errors (ideally is an adverb where you want to use an adjective). For example.

We can see that the cover tree solver is faster than standard standard KNN. In contrast, the kd-tree solver is much slower. This is likely to be caused be the high-dimensional data that we used.